geoportailv3/static/js/query/querydirective.js

```diff
@@ -0,0 +1,242 @@
+          var queryableId = layers[i].get('queryable_id');
+          layersList.push(queryableId);
+          layerLabel[queryableId] = layers[i].get('label');
+        }
+      }
+    }
+    if (layersList.length > 0) {
+      var buffer = 5;
+      var ll = ol.proj.transform(
+          map.getCoordinateFromPixel(
+          [evt.pixel[0] - buffer, evt.pixel[1] - buffer]),
+          map.getView().getProjection(), 'EPSG:2169');
+      var ur = ol.proj.transform(
+          map.getCoordinateFromPixel(
+          [evt.pixel[0] + buffer, evt.pixel[1] + buffer]),
+          map.getView().getProjection(), 'EPSG:2169');
+      var box = ll.concat(ur);
+
+      $http.get(
+          getInfoServiceUrl,
+          {params: {
+            'layers': layersList.join(),
+            'box': box.join()
+          }}).then(
+          goog.bind(function(resp) {
+            goog.array.forEach(resp.data, function(item) {
+              item['layerLabel'] = layerLabel[item.layer];
+            });
+            this['content'] = resp.data;
+            this['infoOpen'] = true;
+            this.clearFeatures_();
+            this.lastHighlightedFeatures_ = [];
+            for (var i = 0; i < resp.data.length; i++) {
+              this.lastHighlightedFeatures_.push.apply(
+                  this.lastHighlightedFeatures_,
+                  resp.data[i].features
+              );
+            }
+            this.highlightFeatures_(this.lastHighlightedFeatures_);
+          },this));
+    }
+  },this));
+
+
+  map.on('pointermove', goog.bind(function(evt) {
+    if (evt.dragging) {
+      return;
+    }
+    var pixel = map.getEventPixel(evt.originalEvent);
+    var hit = map.forEachLayerAtPixel(pixel, function(layer) {
+      if (goog.isDefAndNotNull(layer)) {
+        var metadata = layer.get('metadata');
+        if (goog.isDefAndNotNull(metadata)) {
+          if (goog.isDefAndNotNull(metadata['is_queryable']) &&
+              metadata['is_queryable']) {
+            return true;
+          }
+        }
+      }
+      return false;
+    });
+    map.getTargetElement().style.cursor = hit ? 'pointer' : '';
+  },this));
+};
+
+
+/**
+ * @private
+ */
+app.QueryController.prototype.clearFeatures_ = function() {
+  this.overlay_.getFeatures().clear();
+};
+
+
+/**
+ * provides the template path according with the fact
+ * that the template for the current layer is remote or not
+ * @param {{remote_template: boolean, template: string, layer: string}} layer
+ * @return {string} the template path.
+ * @export
+ */
+app.QueryController.prototype.getTemplatePath = function(layer) {
+  if (layer['remote_template'] === true) {
+    return (this.remoteTemplateUrl_ + '?layer=' + layer['layer']);
+  }
+  return (this.templatePath_ + '/' + layer['template']);
+};
+
+
+/**
+ * @param {Array<string>} features the features to highlight
+ * @private
+ */
+app.QueryController.prototype.highlightFeatures_ = function(features) {
+
+  var encOpt = {
```

Comment:
It would be safer to type this.

You can try to just give it a type first:

```js
/** @type {olx.format.ReadOptions} */
var encOpt = {
  dataProjection: 'EPSG:2169',
  featureProjection: this['map'].getView().getProjection()
};
```

If the compiler does not like it (because your object does not have a `rightHanded` property) then you'll have to use a type-cast:

```js
var encOpt = /** @type {olx.format.ReadOptions} */ ({
  dataProjection: 'EPSG:2169',
  featureProjection: this['map'].getView().getProjection()
});
```

Giving the object a type is safer because it ensures that the compiler will consistently rename object properties.